packages/flutter_tools/lib/src/ios/ios_deploy.dart

```
@@ -169,30 +211,193 @@ class IOSDeploy {
+          // Timed out waiting for the app to launch.
+          _debuggerAttached = false;
+          _logger.printTrace(line);
+          debuggerCompleter.complete(false);
+          return;
+        }
+        if (!debuggerAttached) {
+          _logger.printTrace(line);
+          return;
+        }
+        if (lastLineFromDebugger != null && lastLineFromDebugger.isNotEmpty && line.isEmpty) {
+          // The lldb console stream from ios-deploy is separated lines by an extra \r\n.
+          // To avoid all lines being double spaced, if the last line from the
+          // debugger was not an empty line, skip this empty line.
+          // This will still cause "legit" logged newlines to be doubled...
+        } else {
+          _debuggerOutput.add(line);
+        }
+        lastLineFromDebugger = line;
+      });
+      final StreamSubscription<String> stderrSubscription = _iosDeployProcess.stderr
+          .transform<String>(utf8.decoder)
+          .transform<String>(const LineSplitter())
+          .listen((String line) {
+        _logger.printTrace(line);
+      });
+      unawaited(_iosDeployProcess.exitCode.then((int status) {
+        _logger.printTrace('ios-deploy exited with code $exitCode');
+        _debuggerAttached = false;
+        unawaited(stdoutSubscription.cancel());
+        unawaited(stderrSubscription.cancel());
+      }).whenComplete(() async {
+        if (_debuggerOutput.hasListener) {
+          // Tell listeners the process died.
+          await _debuggerOutput.close();
+        }
+        if (!debuggerCompleter.isCompleted) {
+          debuggerCompleter.complete(false);
+        }
+        _iosDeployProcess = null;
+      }));
+    } on ProcessException catch (exception, stackTrace) {
+      _logger.printTrace('ios-deploy failed: $exception');
+      _debuggerAttached = false;
+      _debuggerOutput.addError(exception, stackTrace);
+    } on ArgumentError catch (exception, stackTrace) {
```

This will fail if ios-deploy is missing right? should we catch the error or let it crash here and solve the problem elsewhere? i.e. startup checks

---

I think we already have those checks in place with the stamp/artifact download logic, since `ios-deploy` is an artifact we ship ourselves `idevicesyslog`.  Unlike the Xcode tooling where we do lots of install checks.